src/main/drivers/bus.c

```diff
@@ -22,11 +22,18 @@
  * along with this program. If not, see http://www.gnu.org/licenses/.
  */
 
+#include <string.h>
+
 #include "platform.h"
+#include "build/debug.h"
 
 #include "drivers/bus.h"
 #include "drivers/io.h"
 
+#define BUSDEV_MAX_DEVICES 8
+
+static busDevice_t busDevPool[BUSDEV_MAX_DEVICES];
```

Comment:
So you are adding dynamic allocation of busDevices. Why? It complicates things and is a potential source of error.

Reply:
I need a busDevice for each active sensor. Initially I did allocate a static `busDevice_t` object  for each **supported** sensor, but that was a waste of memory.

Reply:
Yes, it does waste a bit of memory. But `busDevice_t` is fairly small, and it's not like we have hundreds of sensors. I'm prepared to pay a bit of RAM for increased reliability.

Reply:
Agreed. I'll look into reverting back to static allocation.